dask/dataframe/shuffle.py

```diff
@@ -645,7 +670,13 @@ def set_sorted_index(df, index, drop=True, divisions=None, **kwargs):
     result = map_partitions(M.set_index, df, index, drop=drop, meta=meta)
 
     if not divisions:
-        divisions = compute_divisions(result, **kwargs)
+        divisions, flag = compute_divisions(result, **kwargs)
+
+        if flag:  # need to repartition due to overlap
+            result = fix_overlap(result, divisions)
+            # print (result.compute())
+            # map_partitions(lambda d: print(str(d) + "\n---"), result).compute()
+            result.compute()
```

Comment:
Hopefully we don't need this compute call here.

1.  It's nice to keep everything lazy
2.  If result is large then this would blow up memory

Reply:
Sorry, I accidentally left that in! That was from debugging. I feel so dumb.

Reply:
It's ok.  It turns out that we're all pretty dumb.  Luckily we have others around to help :)